### ISSUE TITLE
Installing tuhi on Ubuntu 18.04

### ISSUE THREAD

**Issue by regtm** [tuhiproject/tuhi#247]
I'm trying to install tuhi on Bionic however i struggle to meet all dependencies. Here is what i tried:

The repo version of meson does not meet the version requirement so i installed it via pip3.
The repo version of pygobject does not meet the requirements so i installed it via pip3 aswell, however building fails, as meson does not find it.

`meson.build:9:0: ERROR: Dependency "pygobject-3.0" not found, tried pkgconfig and cmake`

How can i prepare the pygobject package for use with meson?

Full log:
Build started at 2020-04-04T00:33:16.718740                                                                                                                                                                                                                                                                                                                               
Main binary: /usr/bin/python3                                                                                                                                                                                                                                                                                                                                             
Build Options:                                                                                                                                                                                                                                                                                                                                                            
Python system: Linux                                                                                                                                                                                                                                                                                                                                                      
The Meson build system                                                                                                                                                                                                                                                                                                                                                    
Version: 0.54.0                                                                                                                                                                                                                                                                                                                                                           
Source dir: /home/user/git_repos/tuhi                                                                                                                                                                                                                                                                                                                                     
Build dir: /home/user/git_repos/tuhi/build                                                                                                                                                                                                                                                                                                                                
Build type: native build                                                                                                                                                                                                                                                                                                                                                  
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.                                                                                                                                                                                                                                                                                      
None of 'PKG_CONFIG_PATH' are defined in the environment, not changing global flags.                                                                                                                                                                                                                                                                                      
Project name: tuhi                                                                                                                                                                                                                                                                                                                                                        
Project version: 0.3                                                                                                                                                                                                                                                                                                                                                      
Build machine cpu family: x86_64                                                                                                                                                                                                                                                                                                                                          
Build machine cpu: x86_64                                                                                                                                                                                                                                                                                                                                                 
Host machine cpu family: x86_64                                                                                                                                                                                                                                                                                                                                           
Host machine cpu: x86_64                                                                                                                                                                                                                                                                                                                                                  
Target machine cpu family: x86_64                                                                                                                                                                                                                                                                                                                                         
Target machine cpu: x86_64                                                                                                                                                                                                                                                                                                                                                
Pkg-config binary for MachineChoice.HOST is not cached.                                                                                                                                                                                                                                                                                                                   
None of 'PKG_CONFIG' are defined in the environment, not changing global flags.                                                                                                                                                                                                                                                                                           
Pkg-config binary missing from cross or native file, or env var undefined.                                                                                                                                                                                                                                                                                                
Trying a default Pkg-config fallback at pkg-config                                                                                                                                                                                                                                                                                                                        
Trying pkg-config binary pkg-config for machine MachineChoice.HOST at ['/usr/bin/pkg-config']                                                                                                                                                                                                                                                                             
Found pkg-config: /usr/bin/pkg-config (0.29.1)                                                                                                                                                                                                                                                                                                                            
Determining dependency 'pygobject-3.0' with pkg-config executable '/usr/bin/pkg-config'                                                                                                                                                                                                                                                                                   
PKG_CONFIG_PATH:                                                                                                                                                                                                                                                                                                                                                          
Called `/usr/bin/pkg-config --modversion pygobject-3.0` -> 1                                                                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                                                                          
CMake binary for MachineChoice.HOST is not cached                                                                                                                                                                                                                                                                                                                         
None of 'CMAKE' are defined in the environment, not changing global flags.                                                                                                                                                                                                                                                                                                
CMake binary missing from cross or native file, or env var undefined.                                                                                                                                                                                                                                                                                                     
Trying a default CMake fallback at cmake                                                                                                                                                                                                                                                                                                                                  
Trying CMake binary cmake for machine MachineChoice.HOST at ['/usr/bin/cmake']                                                                                                                                                                                                                                                                                            
Found CMake: /usr/bin/cmake (3.10.2)                                                                                                                                                                                                                                                                                                                                      
Extracting basic cmake information                                                                                                                                                                                                                                                                                                                                        
Try CMake generator: auto                                                                                                                                                                                                                                                                                                                                                 
None of 'CC' are defined in the environment, not changing global flags.                                                                                                                                                                                                                                                                                                   
None of 'CFLAGS' are defined in the environment, not changing global flags.                                                                                                                                                                                                                                                                                               
None of 'LDFLAGS' are defined in the environment, not changing global flags.                                                                                                                                                                                                                                                                                              
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CC_LD' are defined in the environment, not changing global flags.
None of 'CXX' are defined in the environment, not changing global flags.
None of 'CXXFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'CPPFLAGS' are defined in the environment, not changing global flags.
None of 'CXX_LD' are defined in the environment, not changing global flags.
None of 'FC' are defined in the environment, not changing global flags.
None of 'FFLAGS' are defined in the environment, not changing global flags.
None of 'LDFLAGS' are defined in the environment, not changing global flags.
None of 'FC_LD' are defined in the environment, not changing global flags.
None of 'F_LD' are defined in the environment, not changing global flags.
Called `/usr/bin/cmake --trace --trace-expand --no-warn-unused-cli .` in /home/user/git_repos/tuhi/build/meson-private/cmake_pygobject-3.0 -> 0
  -- Module search paths:    ['/', '/opt', '/usr', '/usr/local']
  -- CMake root:             /usr/share/cmake-3.10
  -- CMake architectures:    ['x86_64-linux-gnu']
  -- CMake lib search paths: ['lib', 'lib32', 'lib64', 'libx32', 'share', 'lib/x86_64-linux-gnu']
None of 'CMAKE_PREFIX_PATH' are defined in the environment, not changing global flags.
Preliminary CMake check failed. Aborting.
Run-time dependency pygobject-3.0 found: NO (tried pkgconfig and cmake)

meson.build:9:0: ERROR: Dependency "pygobject-3.0" not found, tried pkgconfig and cmake

**Comment by whot** [comment]
I don't know what version of gobject 18.04 ships and I haven't touched Ubuntu in years, sorry. Note that python-gobject is a wrapper around gobject the C library, so I don't think installing via pip3 is sufficient here. Beyond that - I don't know. Probably best to run the Tuhi flatpak?

**Comment by kir0ul** [comment]
I have the exact same problem... :slightly_frowning_face:

**Comment by HoBoIs** [comment]
I had that too I fixed it by editing "/usr/lib/x86_64-linux-gnu/pkgconfig/pygobject-3.0.pc" 
It became:
[pygobj.txt](https://github.com/tuhiproject/tuhi/files/4975583/pygobj.txt)
note that USERNAME should be your username 
(It exsisted becouse an older pygobject was installed from pacetmanagger. I changed version and prefix)

**Comment by ericzolf** [comment]
Same issue under Fedora 32, I haven't tried HoBols' workaround because I haven't yet understood it.

**Comment by ericzolf** [comment]
Installing the packages `python3-gobject-devel` brought me forward and installing `python3-svgwrite` made the build succeed, addressing the follow-up error (definitely easier to interpret):

```
Program python3 (svgwrite, xdg, gi, cairo) found: NO modules: xdg, gi, cairo

meson.build:40:4: ERROR: python is missing modules: svgwrite
```

A better documentation of the dependencies would have helped.

**Comment by whot** [comment]
@ericzolf  I can give you wiki access if you want, that's usually the best place to document these dependencies. having said that "python is missing module" is a standard meson message, I'd hope that there is some generic documentation out there. For the .pc files I wrote [this post](https://who-t.blogspot.com/2018/07/meson-fails-with-native-dependency-not-found.html) years ago.

**Comment by PhoenixSmaug** [comment]
I got it running on Linux Mint 20 after several hours of trying, it should be a similar process for Ubuntu (Virtual Python Environment optional):
```
git clone http://github.com/tuhiproject/tuhi
cd tuhi

python3 -m venv venv
source venv/bin/activate

pip install meson
sudo apt install pkg-config
sudo apt install libcairo2-dev
pip install wheel
sudo apt install gcc
sudo apt install libgirepository1.0-dev
sudo apt install python3-dev
pip install pycairo
pip install pygobject
pip install svgwrite
pip install xdg
pip install pyxdg
sudo apt install python-gi-dev
sudo apt install gettext
pip install ninja

meson builddir
ninja -C builddir
./builddir/tuhi.devel
```

That's how I got it working, but it's not impossible that single installations are redundant.

**Comment by robol** [comment]
In case anybody is still trying to install tuhi on Ubuntu 18.04, I created debian packages (also for 20.04) here: https://launchpad.net/~leo.robol/+archive/ubuntu/tuhi. They ship tuhi-live.py as well because it worked reliably for me.

To run meson on Ubuntu 18.04 I just lowered the minimum required version of pygobject to 3.26, the one present in Ubuntu Bionic. I honestly have no idea if some features present in 3.28 or 3.30 are used in the code, but after some testing it appears to be working correctly.
